data_model/model.py

```diff
@@ -0,0 +1,244 @@
+
+    def get_rooms_for_house(self, rooms, house_id):
+        lst = []
+        for room in rooms:
+            if room.house_id == house_id:
+                lst += room
+        return lst
+
+
+class Device(Room):
+    def __init__(self, house_id, room_id, device_id, name, device_type, power_state, last_temp, target_temp,
+                 sensor_data):
+        self.house_id = house_id
+        self.room_id = room_id
+        self.device_id = device_id
+        self.name = name
+        self.device_type = device_type
+        self.power_state = power_state
+        self.last_temp = last_temp
+        self.target_temp = target_temp
+        self.sensor_data = sensor_data
+
+    def get_device_attributes(self):
+        return {'house_id': self.house_id, 'room_id': self.room_id, 'device_id': self.device_id, 'name': self.name,
+                'device_type': self.device_type, 'power_state': self.power_state, 'last_temp': self.last_temp,
+                'target_temp': self.target_temp, 'sensor_data': self.sensor_data}
+
+    def change_power_state(self):
+        try:
+            if self.power_state == 1:
+                self.power_state = 0
+            elif self.power_state == 0:
+                self.power_state = 1
+        except:
+            return "ERROR: Power State not 1 or 0"
+
+    def set_target_temp(self, target):
+        self.target_temp = target
+
+
+
+class DeviceRepository:
+    def __init__(self):
+        self.devices = {}
+
+    def add_device(self, device):
+        self.devices[device.id] = device
+
+    def remove_device(self, device_id):
+        self.devices.pop(device_id, None)
+
+    def get_device_by_id(self, device_id):
+        try:
+            return self.devices[device_id]
+        except KeyError:
+            print("Device %d not found" % device_id)
+
+    def add_device_to_house(self, house, device_id):
+        house += self.devices[device_id]
+
+    def get_devices_for_house(self, devices, house_id):
+        lst = []
+        for device in devices:
+            if device.house_id == house_id:
+                lst += device
+        return lst
+
+    def add_device_to_room(self, room, device_id):
+        room += self.devices[device_id]
+
+    def get_devices_for_room(self, devices, room_id):
+        lst = []
+        for device in devices:
+            if device.room_id == room_id:
+                lst += device
+        return lst
+
+
+class DeviceGroups(object):
+    def __init__(self, device_id, device_group_id):
+        self.device_id = device_id
+        self.device_group_id = device_group_id
+
+    def get_device_group_attributes(self):
+        return {'device_id': self.device_id, 'device_group_id': self.device_group_id}
+
+
+class DeviceGroupRepository:
+    def __init__(self):
+        self.device_groups = {}
+
+    def add_device_group(self, device_group):
+        self.device_groups[device_group.id] = device_group
+
+    def add_device_to_group(self, device_id, device_group):
```

Comment:
Try to be consistent when choosing the ordering of arguments and when to ask for the ID of an object or rather the object itself.